featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -299,7 +300,7 @@ def _run_dfs(self, entity, entity_path, all_features, max_depth):
         """
 
         backward_entities = self.es.get_backward_entities(entity.id)
-        backward_entities = [b_id for b_id in backward_entities
+        backward_entities = [b_id for b_id, _ in backward_entities
```

Comment:
Now that we have the concept of `RelationshipPath`, perhaps it makes sense to keep the relationship path here, and refactor the entity_path argument to use it rather than just a list of entities. 

the entity path is basically used to keep track of how we've travserved to a given entity, so the relationship path would be a natural thing to use instead.